heron/scheduler-core/src/java/com/twitter/heron/scheduler/UpdateTopologyManager.java

```diff
@@ -104,12 +126,124 @@ public void updateTopology(final PackingPlans.PackingPlan existingProtoPackingPl
     // delete the physical plan so TMaster doesn't try to re-establish it on start-up.
     logFine("Deleted Physical Plan: %s", stateManager.deletePhysicalPlan(topologyName));
 
-    if (removableContainerCount > 0) {
+    // reactivate topology
+    reactivateTopology(stateManager, updatedTopology, removableContainerCount);
+
+    if (removableContainerCount > 0 && scalableScheduler.isPresent()) {
       scalableScheduler.get().removeContainers(containerDelta.getContainersToRemove());
     }
   }
 
   @VisibleForTesting
+  void deactivateTopology(SchedulerStateManagerAdaptor stateManager,
+                          final TopologyAPI.Topology topology)
+      throws InterruptedException {
+
+    List<TopologyAPI.Config.KeyValue> topologyConfig = topology.getTopologyConfig().getKvsList();
+    long deactivateSleepSeconds = TopologyUtils.getConfigWithDefault(
```

Comment:
That is the intent yes, which is that the setting can be configured at topology submission time in the main method, or overridden at update time from the command line.